### PR TITLE
Replace RadIO with TorchIO for patch-based inference

### PR DESCRIPTION
**Pull request by fepegar**
Write a detailed description of your PR here.

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [ ] Ensure that your PR is small, and implements one change.
- [ ] Add unit tests for all functions that you introduced or modified.
- [ ] Run PyCharm's code cleanup tools on your Python files.
- [ ] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [ ] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

**Comment by fepegar** [comment]
I "fixed" some mypy errors by explicitly ignoring them. I'm not sure this is the best solution.

File: InnerEye/README.md [hunk 1]
```diff
@@ -15,7 +15,6 @@ dependencies:
   - python=3.7.3
   - pytorch=1.3.0
   - pip:
-      - git+https://github.com/analysiscenter/radio.git@6d53e25#egg=radio
       - git+https://github.com/ptrblck/apex.git@4ad9b3b#egg=apex
```
Comment (javier-alvarez) [inline InnerEye/README.md:18]: remove apex also please from this example. This was before PL

**Review by ant0nsc — APPROVED** [review]
Looking GREAT! Lots of ugly code replaced by uncle tio ;-)

**Review by javier-alvarez — CHANGES_REQUESTED** [review]
https://github.com/microsoft/InnerEye-DeepLearning/issues/665 Lets address this one before merging and run manual regression test on prostate and H&N

**Comment by fepegar** [comment]
# Comparison RadIO vs TorchIO

I suspect the tiny differences might be due to implementation details. For example, if an image needs to be 3 voxels larger along a specific dimensions, we could add 2 voxels at the beginning and 1 at the end, or vice versa.

## Quantitative

### RadIO

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>Structure</th>
      <th>25%</th>
      <th>50%</th>
      <th>75%</th>
      <th>count</th>
      <th>max</th>
      <th>mean</th>
      <th>min</th>
      <th>std</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>external</td>
      <td>0.992</td>
      <td>0.996</td>
      <td>0.997</td>
      <td>109.0</td>
      <td>0.998</td>
      <td>0.994706</td>
      <td>0.969</td>
      <td>0.003560</td>
    </tr>
    <tr>
      <td>femur_l</td>
      <td>0.978</td>
      <td>0.980</td>
      <td>0.982</td>
      <td>109.0</td>
      <td>0.988</td>
      <td>0.979541</td>
      <td>0.962</td>
      <td>0.003938</td>
    </tr>
    <tr>
      <td>femur_r</td>
      <td>0.978</td>
      <td>0.980</td>
      <td>0.981</td>
      <td>109.0</td>
      <td>0.987</td>
      <td>0.979486</td>
      <td>0.967</td>
      <td>0.003760</td>
    </tr>
    <tr>
      <td>bladder</td>
      <td>0.960</td>
      <td>0.971</td>
      <td>0.979</td>
      <td>109.0</td>
      <td>0.990</td>
      <td>0.965761</td>
      <td>0.840</td>
      <td>0.023162</td>
    </tr>
    <tr>
      <td>rectum</td>
      <td>0.856</td>
      <td>0.899</td>
      <td>0.921</td>
      <td>109.0</td>
      <td>0.961</td>
      <td>0.886587</td>
      <td>0.636</td>
      <td>0.048407</td>
    </tr>
    <tr>
      <td>prostate</td>
      <td>0.823</td>
      <td>0.867</td>
      <td>0.903</td>
      <td>109.0</td>
      <td>0.954</td>
      <td>0.853202</td>
      <td>0.629</td>
      <td>0.068248</td>
    </tr>
    <tr>
      <td>seminalvesicles</td>
      <td>0.758</td>
      <td>0.799</td>
      <td>0.841</td>
      <td>109.0</td>
      <td>0.909</td>
      <td>0.785954</td>
      <td>0.189</td>
      <td>0.089569</td>
    </tr>
  </tbody>
</table>

### TorchIO

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>Structure</th>
      <th>25%</th>
      <th>50%</th>
      <th>75%</th>
      <th>count</th>
      <th>max</th>
      <th>mean</th>
      <th>min</th>
      <th>std</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>external</td>
      <td>0.992</td>
      <td>0.996</td>
      <td>0.997</td>
      <td>109.0</td>
      <td>0.998</td>
      <td>0.994642</td>
      <td>0.969</td>
      <td>0.003591</td>
    </tr>
    <tr>
      <td>femur_l</td>
      <td>0.978</td>
      <td>0.980</td>
      <td>0.982</td>
      <td>109.0</td>
      <td>0.988</td>
      <td>0.979596</td>
      <td>0.962</td>
      <td>0.003923</td>
    </tr>
    <tr>
      <td>femur_r</td>
      <td>0.978</td>
      <td>0.980</td>
      <td>0.982</td>
      <td>109.0</td>
      <td>0.988</td>
      <td>0.979596</td>
      <td>0.966</td>
      <td>0.003847</td>
    </tr>
    <tr>
      <td>bladder</td>
      <td>0.960</td>
      <td>0.972</td>
      <td>0.978</td>
      <td>109.0</td>
      <td>0.990</td>
      <td>0.965532</td>
      <td>0.824</td>
      <td>0.024132</td>
    </tr>
    <tr>
      <td>rectum</td>
      <td>0.857</td>
      <td>0.899</td>
      <td>0.916</td>
      <td>109.0</td>
      <td>0.963</td>
      <td>0.884596</td>
      <td>0.630</td>
      <td>0.050930</td>
    </tr>
    <tr>
      <td>prostate</td>
      <td>0.822</td>
      <td>0.864</td>
      <td>0.905</td>
      <td>109.0</td>
      <td>0.953</td>
      <td>0.853560</td>
      <td>0.612</td>
      <td>0.067435</td>
    </tr>
    <tr>
      <td>seminalvesicles</td>
      <td>0.758</td>
      <td>0.800</td>
      <td>0.840</td>
      <td>109.0</td>
      <td>0.913</td>
      <td>0.786560</td>
      <td>0.155</td>
      <td>0.089954</td>
    </tr>
  </tbody>
</table>

## Qualitative

ezgif-4-5713aa4c9e

**Comment by fepegar** [comment]
I can run H&N too, but I suspect results will be very similar again.

**Review by javier-alvarez — APPROVED** [review]

**Pull request merged**